tests/h/search/query_test.py

```diff
@@ -391,17 +391,22 @@ def search(self, search):
         return search
 
 
-class TestUriFilter(object):
+class TestUriCombinedWildcardFilter():
+    # TODO - Explicit test of URL normalization (ie. that search normalizes input
+    # URL using `h.util.uri.normalize` and queries with that).
+
     @pytest.mark.parametrize("field", ("uri", "url"))
-    def test_filters_by_field(self, search, Annotation, field):
+    def test_filters_by_field(self, search, Annotation, pyramid_request, field):
+        search = self._get_search(search, pyramid_request, True)
```

Comment:
There is some duplication across the test cases in this class in the initialization of `search` by calling `_get_search`. It could perhaps be simplified a little with something like:

```py
def test_something(get_search):
  search = get_search()

def test_something(get_search):
  search = get_search(separate_keys=False)

@pytest.fixture
def get_search(search, pyramid_request):
  def _get_search(separate_keys=True):
    search.append_modifier(...)
    return search
return _get_search
```

Reply:
It does look a lot cleaner now. 💯